src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -330,6 +346,8 @@ def _(self, node: QuantumGate) -> None:
         gate_name = node.name.name
         arguments = self.visit(node.arguments)
         modifiers = self.visit(node.modifiers)
+        if self.context.in_global_scope and modifiers:
+            self._uses_advanced_language_features = True
```

Comment:
Checking global scope, since builtin gates may have gate modifiers in their definitions. Though this leaves a hole for using a gate modifier outside of global scope, ie inside a for loop, other ways of getting out of global scope would trigger the advanced language features flag anyways